src/storyblok/components/ProjectList.astro

```diff
@@ -1,48 +1,164 @@
 ---
+import iconStampTools from '../../images/svgs/stamp-tools.svg?raw';
+import iconStampExperiments from '../../images/svgs/stamp-experiments.svg?raw';
+import iconStampStunts from '../../images/svgs/stamp-stunts.svg?raw';
 import Image from './Image.astro';
+import ToggleButton from './ToggleButton.astro';
+import Toggle from './Toggle.astro';
+
+function getIconStamp(type) {
+  switch (type) {
+    case 'tools':
+      return iconStampTools;
+    case 'experiments':
+      return iconStampExperiments;
+    case 'stunts':
+      return iconStampStunts;
+    default:
+      return '';
+  }
+}
+
+function getContrastColor(hexColor) {
+  // Remove the hash at the start if it's there
+  hexColor = hexColor.replace(/^#/, '');
+
+  // Parse the color to RGB
+  const r = parseInt(hexColor.substring(0, 2), 16);
+  const g = parseInt(hexColor.substring(2, 4), 16);
+  const b = parseInt(hexColor.substring(4, 6), 16);
+
+  // Calculate the perceptive luminance (aka luma) - human eye favors green color
+  const luma = ((0.299 * r) + (0.587 * g) + (0.114 * b)) / 255;
+
+  // Return black for bright colors, white for dark colors
+  return luma > 0.5 ? '#000' : '#FFF';
+}
+
 const { projects } = Astro.props;
 ---
 
-<ul class="grid grid-cols-3 gap-5">
-  {projects.map((project) => {
-    const { content } = project;
-    const backgroundColor = content.backgroundColor?.color || '#000';
-    const textColor = content.textColor?.color || '#fff';
-
-    return (
-      <li
-        class="project-item"
-        data-type={content.type}
-        style={`
-          --projectBackgroundColor: ${backgroundColor};
-          --projectTextColor: ${textColor};
-        `}
-      >
-        <a href={`/${project.slug}`}>
-          {content.icon && <Image image={content.icon} />}
-
-          <h3>{project.name}</h3>
-
-          {content.tagline && <p>{content.tagline}</p>}
-        </a>
-      </li>
-    );
-  })}
-</ul>
+<div
+  class="px-3"
+  x-data="{
+    showMore: $persist(false),
+    filters: $persist({
+      tools: true,
+      experiments: true,
+      stunts: true
+    }),
+    projects: JSON.parse(JSON.stringify(projects)),
```

Comment:
I'm not sure if I'm seeing how `projects` is used within Alpine. 

I see astro looping over projects on line 111, but I don't think that's making use of what's here in Alpine.

Reply:
👍 I must have forgotten to remove this when refactoring. Thanks for the catch!